### ISSUE TITLE
Clear input X icon is not keyboard accessible on search component

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/vets-design-system-documentation#2696]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

On the [Search input component](https://design.va.gov/storybook/?path=/docs/uswds-va-search-input--default) I was using my keyboard to navigate and I was not able to access the X icon to clear my input

## What I expected to happen

I should be able to access the X icon with the keyboard and then activate it to clear the input in the search component just as I would when using my mouse.

## Reproducing
- Formation version:
- Device: (e.g. iPhone 8, Macbook) Mac
- Browser: (e.g. Firefox, IE 11) Chrome

Steps to reproduce:

1. Type a term into the search component
2. Using the keyboard, tab through the elements
3. Notice that you can tab to and focus on the input field and the search button. The X icon is skipped over.

## Urgency
How urgent is this request? Please select the approriate option below and/or provide details
- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [x] Other

### Details
I also tested this on USWDS and noticed that this occurs there as well.  After some investigation, Dan and Jami discovered that this is a browser "bug", not USWDS or VADS code issue. You can see the same issue on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/search) example

## Estimating
Provide your estimate of 1, 2, 3, 5, 8 or 13 
@Andrew565 - 3
@ataker - 3
@harshil1793 -  3
@it-harrison - 3
@jamigibbs -
@micahchiang - 3
@powellkerry - 3
@rmessina1010 -
@rsmithadhoc - 3

**Comment by humancompanion-usds** [comment]
@rsmithadhoc - Please review this and set a defect level label on this issue. Thanks!

**Comment by caw310** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/2696?planning-poker) with Zenhub @Andrew565 @ataker @harshil1793 @it-harrison @jamigibbs @micahchiang @powellkerry @rmessina1010 @rsmithadhoc

**Comment by jamigibbs** [comment]
The search component as built by USWDS uses the browser native `type=search` on the `input` element:

**Comment by jamigibbs** [comment]
The search component as built by USWDS using the browser native `type=search` on the `input` element. 

Keyboard access to clear the value is done using the `Esc` key on the keyboard. I'm not sure that we can control tabbing into that X because the behavior is up to the user agent.

**Comment by humancompanion-usds** [comment]
@laflannery - Was your question to the USWDS team around [this issue](https://github.com/uswds/uswds/issues/5277) related to this request?

**Comment by laflannery** [comment]
@humancompanion-usds I did ask them about that issue as well:
- They confirmed, as you all did that this is a browser issue so the behavior and therefore the accessibility issues are not uniform across browsers and devices
- They determined not to prioritize a fix at this time
- I also asked if they had any research on this component and if there was any data to determine if this was presenting any issues or not - unfortunately there wasn't user research yet

So if, based on this, it would be more efficient to close or hold this until there's a bit more from the USWDS side with everything else on your plate that makes sense.

**Comment by Andrew565** [comment]
One possible option we could implement to make this more screen reader friendly would be to add "Press the 'esc' key to clear the input" or a similar message to the sr-only div?

A second, much more intensive option would be to completely rewrite the component to make use of a standard "text" input (instead of the "search" input) and then create our own "x" that was keyboard-focusable.

**Comment by caw310** [comment]
Removing a11y-defect-1 label since there is a work around.

**Comment by rsmithadhoc** [comment]
An implementation of a custom X to clear button is present on the [Combo box](https://design.va.gov/storybook/?path=/docs/uswds-va-combo-box--docs) that was implemented by another team, so that may be a good example to look at for this.

**Comment by caw310** [comment]
Closing as this is a duplicate of in https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3590 which will be worked on early Jan.
